lib/src/formatter.dart

```diff
@@ -166,17 +166,17 @@ abstract class ReportFormatter {
 class SimpleFormatter implements ReportFormatter {
   final IOSink out;
   final Iterable<AnalysisErrorInfo> errors;
-  final LintFilter filter;
+  final LintFilter? filter;
 
   int errorCount = 0;
   int filteredLintCount = 0;
 
-  final int fileCount;
-  final int elapsedMs;
-  final String fileRoot;
+  final int? fileCount;
+  final int? elapsedMs;
+  final String? fileRoot;
   final bool showStatistics;
-  final bool machineOutput;
-  final bool quiet;
+  final bool? machineOutput;
+  final bool? quiet;
```

Comment:
`bool?` is a strong signal that there's a bug in the code. Consider making these non-nullable and clean up the code.